BattleShipsClient/BattleShipsClient/Menu.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using System.Windows.Forms;
+using System.Net;
+using System.Net.Sockets;
+
+namespace BattleShipsClient
+{
+    public partial class Menu : Form
+    {
+        public Form1 f1 = new Form1();
+        TcpClient client = new TcpClient();
+        public bool first = true;
+
+        public Menu()
+        {
+            InitializeComponent();
+        }
+        private void Menu_Load(object sender, EventArgs e)
+        {
+            if (first == true)
+            {
+                Program.MakeLog();
+                Con();
+            }
+            else
+            {
+                Thread rd = new Thread(f1.recievedata);
+                rd.Start();
+                f1 = new Form1();
+                f1.client = client;
+                f1.menu = this;
+                f1.Send("UN:" + Environment.UserName);
+                this.Text = "Menu";
+                JoinPNL.Show();
+            }
+            Refresh_Click(null, EventArgs.Empty);
+        }
+        public void Con()
+        {
+            bool valid = IPAddress.TryParse(Properties.Resources.IPAdress, out IPAddress ipaddress);
+            client.Connect(ipaddress, 666);
```

Comment:
You may want to consider what happens if it fails to connect: how long will it wait to timeout? what happens when it does?
This will also freeze the form while it connects, which may not be user-friendly